packages/@aws-cdk/aws-iam/lib/account-utils.ts

```diff
@@ -0,0 +1,23 @@
+import * as cdk from '@aws-cdk/core';
+
+export enum AccountCompare {
+  SAME,
+  DIFFERENT,
+  UNKNOWN,
+}
+
+export function accountsAreSameOrUnresolved(account1: string | undefined, account2: string | undefined): AccountCompare {
+  const firstIsUnresolved = cdk.Token.isUnresolved(account1);
+  const secondIsUnresolved = cdk.Token.isUnresolved(account2);
+
+  if (firstIsUnresolved && secondIsUnresolved) {
+    return AccountCompare.SAME;
```

Comment:
This assumes that if account is unresolved it will definitely resolve to the pseudo reference `AWS::Account` but it could also be a lazy value for example.

Maybe we should be a bit smarter about this and allow you to “peek” into the token and only consider them “same” if they both resolve to the pseudo.

Reply:
You're correct, but do you have an idea how we can do this "peeking"?